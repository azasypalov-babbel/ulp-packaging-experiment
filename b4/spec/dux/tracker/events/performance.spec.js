import getPerformanceTrackingMetrics from '../../../../src/lib/performanceTrackingMetrics';
import {
  performancePageLoadedEvent
} from '../../../../src/dux/tracker/events/performance';

jest.mock('../../../../src/lib/performanceTrackingMetrics');

describe('performance events', () => {
  describe('performance:page_loaded', () => {
    beforeEach(() => {
      getPerformanceTrackingMetrics.mockImplementation(() => ({
        connect_end: 1559831288237,
        connect_start: 1559831288237,
        dom_complete: 1559831290414,
        dom_content_loaded_event_end: 1559831289480,
        dom_content_loaded_event_start: 1559831289323
      }));
    });

    test('performance:page_loaded', () => {
      expect(
        performancePageLoadedEvent({ locale: 'en', learnLanguageAlpha3: 'DEU' })
      ).toMatchSnapshot();
    });
  });
});
