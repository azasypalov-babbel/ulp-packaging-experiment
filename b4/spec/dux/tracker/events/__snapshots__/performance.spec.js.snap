// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performance events performance:page_loaded performance:page_loaded 1`] = `
Object {
  "event": "performance:page_loaded",
  "payload": Object {
    "connect_end": 1559831288237,
    "connect_start": 1559831288237,
    "dom_complete": 1559831290414,
    "dom_content_loaded_event_end": 1559831289480,
    "dom_content_loaded_event_start": 1559831289323,
    "learn_language_alpha3": "DEU",
    "locale": "en",
    "provider": "lesson_player",
  },
  "version": 1,
}
`;
